### ISSUE TITLE
Documentation about no_ack key is incorrect

### ISSUE THREAD

**Issue by horgh** [Tamarou/Crixa#11]
Hello,

I noticed that the behaviour of `no_ack` is the opposite of how the documentation describes it. Its description on `Crixa::Queue` says:

>If this is true, then the message is not acknowledged as it is taken from the queue. You will need to explicitly acknowledge it using the ack method on the Crixa::Channel  object from which the message came.
>If this is false, then the message is acknowledged immediately. Calling the ack method later with this message's delivery tag will be an error.

However, I've found that if it's true then the message is acknowledged. If it's false, it is not.

Here is a sample program showing this:

```
# cat crixa.pl 
use strict;
use warnings;

use Crixa ();

my $mq = Crixa->connect( host => 'localhost' );

die 'could not connect' unless $mq->is_connected;

my $channel = $mq->new_channel;

my $exchange = $channel->exchange( name => 'exchangename' );

my $queue_name = 'testqueue';

my $queue = $exchange->queue({
    name         => $queue_name,
    routing_keys => [ $queue_name ],
});

$exchange->publish({
    routing_key => $queue_name,
    body        => 'Hello World'
});

$queue->consume(
    sub {
        return 0 unless @_;
        my $m = shift @_;
        print 'received message with body: ', $m->body, "\n";
        $channel->ack($m->delivery_tag);
        return 1;
    },
    timeout => 1000,
    no_ack  => 1,
);
# perl crixa.pl 
received message with body: Hello World
recv: server channel error 406, message: PRECONDITION_FAILED - unknown delivery tag 1 at /usr/local/share/perl/5.24.1/Crixa/Queue.pm line 148.
```

**Comment by autarch** [comment]
All Crixa is doing is passing this along to the underlying `Net::AMQP::RabbitMQ` object's consume method. I'm pretty sure I wrote code in the past that used this, so I'm not sure what's going on here.

**Comment by autarch** [comment]
I would suggest testing against `Net::AMQP::RabbitMQ`. I strongly suspect that if there is a bug it's in that module. Crixa is just an OO-ier wrapper around it.

**Comment by horgh** [comment]
Thanks for replying!

As you suggested I've tested this with `Net::AMQP::RabbitMQ` (see script [here](https://gist.github.com/horgh/110b57e63d6ce55c11392b91697eceec)).

The behaviour of the flag is the same:

```
Unbinding queue: server channel error 406, message: PRECONDITION_FAILED - unknown delivery tag 1 at net-amqp-rabbitmq.pl line 47.
```

After looking at this some more, I do think `Crixa`'s documentation needs updating.

`Net::AMQP::RabbitMQ` does not state the behaviour of `no_ack`. This leaves us looking for what librabbitmq says about this.

Looking at `amqp.h` from `rabbitmq-c`, `amqp_basic_get()` says this about `no_ack`:

> if true the message is automatically ack'ed
> if false amqp_basic_ack should be called once the message retrieved has been processed

I believe the rationale for `no_ack` is to tell the server "we won't be acking this". The [Erlang client docs](http://www.rabbitmq.com/erlang-client-user-guide.html) support this idea:

> Note that the previous example sets the no_ack flag on the #'basic.get'{} command. This tells the broker that the receiver will not send an acknowledgement of the message. In doing so, the broker can absolve itself of the responsibility for delivery - once it believes it has delivered a message.

**Comment by autarch** [comment]
Ah, I see. This is a lesson about naming anything "no_*". So confusing!

I'm not really using Crixa these days, but a PR to fix this would be very welcome.

**Comment by perigrin** [comment]
Fixed with the merged #12
